transaction_benchmarks.rs fails to compile: Database::storage() is pub(crate)

## Description

`cargo bench --bench transaction_benchmarks -p strata-engine` fails to compile:

```
error[E0624]: method `storage` is private
   --> crates/engine/benches/transaction_benchmarks.rs:268:29
    |
268 |             let result = db.storage().get(&get_key);
    |                             ^^^^^^^ private method
```

## Root Cause

`Database::storage()` was changed to `pub(crate)` visibility, but `bench_direct_operations` in `transaction_benchmarks.rs` calls it directly. Benchmarks are external to the crate and cannot access `pub(crate)` methods.

## Fix

In `crates/engine/benches/transaction_benchmarks.rs`, the `bench_direct_operations` function (line 236) needs to be updated to use the public transaction API instead of calling `db.storage().get()` directly. For example, replace:

```rust
let result = db.storage().get(&get_key);
```

with a `db.transaction()` call that performs the get.

--- a/crates/engine/benches/transaction_benchmarks.rs
+++ b/crates/engine/benches/transaction_benchmarks.rs
@@ -11,7 +11,6 @@ use std::sync::Arc;
 use std::thread;
 use strata_core::types::{BranchId, Key, Namespace};
 use strata_core::value::Value;
-use strata_core::Storage;
 use strata_engine::Database;
 use tempfile::TempDir;
 
@@ -265,7 +264,7 @@ fn bench_direct_operations(c: &mut Criterion) {
 
     group.bench_function("direct_get", |b| {
         b.iter(|| {
-            let result = db.storage().get(&get_key);
+            let result = db.transaction(branch_id, |txn| txn.get(&get_key));
             black_box(result.unwrap());
         });
     });

